### PR TITLE
Music: validation callout

### PR DESCRIPTION
**Pull request by breville**
This allows a validation to specify a callout to be shown at the same time as the validation appears in the instructions panel.

### Screen recording

https://github.com/user-attachments/assets/a11400ed-8605-4210-a418-ad56f246cd5c

### Levelbuilder

Screenshot 2024-09-27 at 11 02 17 PM

### CODE REVIEW

File: apps/src/lab2/levelEditors/validations/EditValidation.tsx [hunk 1]
```diff
@@ -94,6 +94,22 @@ const EditValidation: React.FunctionComponent<EditValidationProps> = ({
           }}
         />
       </div>
+      <div className={moduleStyles.row}>
+        <label htmlFor="callout" className={moduleStyles.label}>
+          Callout (optional):
+        </label>
+        <textarea
```
Comment (sanchitmalhotra126) [inline apps/src/lab2/levelEditors/validations/EditValidation.tsx]: super nit: can this just be an `input` since it's a single line?

**Review by sanchitmalhotra126 — APPROVED** [review]

File: apps/src/lab2/levelEditors/validations/EditValidation.tsx [hunk 2]
```diff
@@ -94,6 +94,22 @@ const EditValidation: React.FunctionComponent<EditValidationProps> = ({
...
+          id="callout"
+          name="callout"
+          rows={1}
+          className={moduleStyles.callout}
+          value={validation.callout}
+          onChange={e => {
+            validation.callout = e.target.value;
```
Comment (levadadenys) [inline apps/src/lab2/levelEditors/validations/EditValidation.tsx:108]: Modifying props object seems tricky to me, are we sure it'll always work as expected? (Keeping in mind possible react/js magic)
Maybe rewrite like this?
`e => onValidationChange({...validation, callout: e.target.value})`

File: apps/src/lab2/levelEditors/validations/EditValidation.tsx [hunk 3]
```diff
@@ -94,6 +94,22 @@ const EditValidation: React.FunctionComponent<EditValidationProps> = ({
           }}
         />
       </div>
+      <div className={moduleStyles.row}>
+        <label htmlFor="callout" className={moduleStyles.label}>
+          Callout (optional):
+        </label>
+        <textarea
+          id="callout"
```
Comment (levadadenys) [inline apps/src/lab2/levelEditors/validations/EditValidation.tsx]: nit: This id looks generic to me, afraid of possible collisions in the future. Maybe use something more specific like `levelEditors_editValidation_callout`?

**Review by levadadenys — APPROVED** [review]
Couple small comments, otherwise LGTM!

File: apps/src/lab2/levelEditors/validations/EditValidation.tsx (hunk 1 above)
Comment (breville) [reply apps/src/lab2/levelEditors/validations/EditValidation.tsx]: Updated.

File: apps/src/lab2/levelEditors/validations/EditValidation.tsx (hunk 3 above)
Comment (breville) [reply apps/src/lab2/levelEditors/validations/EditValidation.tsx]: Updated.  (Though not mixing camel case & snake case, which worries me. :)

File: apps/src/lab2/levelEditors/validations/EditValidation.tsx (hunk 2 above)
Comment (breville) [reply apps/src/lab2/levelEditors/validations/EditValidation.tsx:108]: Oh, good question.  I've left as-is for now to be consistent with the rest of the file, but we could consider updating all of them separately.

**Pull request merged**